lib/pipeline.js

```diff
@@ -1021,6 +1029,11 @@ class PipelineModel extends BaseModel {
                 // Handle internal nodes
                 if (job) {
                     node.id = job.id;
+                    // Add stageId
+                } else if (/^(?:~)?(stage@)/.test(node.name)) {
+                    const stage = await stageFactory.get({ params: { pipelineId: this.id, name: node.name } });
+
+                    node.stageId = stage.id;
```

Comment:
This would throw an error, when stage does not exist in DB.
Creating/Updating stages should be done during PR sync to be consistent with how we handle rest of the pipeline configuration.
Related to https://github.com/screwdriver-cd/models/pull/570#discussion_r1238622719

Reply:
Also should make the node type explicit. Instead of implicitly identifying a node with `id` attribute as a job and a node with `stageId` as a stage. Something like
```
node.id = stage.id;
node.type = 'STAGE';
```